skimage/segmentation/_slic.pyx

```diff
@@ -78,8 +82,8 @@ def _slic_cython(double[:, :, :, ::1] image_zyx,
     step_z, step_y, step_x = [int(s.step if s.step is not None else 1)
                               for s in slices]
 
-    cdef Py_ssize_t[:, :, ::1] nearest_segments \
-        = np.empty((depth, height, width), dtype=np.intp)
+    cdef int[:, :, ::1] nearest_segments \
+        = np.full((depth, height, width), -1, dtype=np.int32)
```

Comment:
Why did you change this dtype?

Reply:
@jni , first of all, thank you for your review. I tried to stick as much as possible to the reference implementation, so some of the implementation decisions come from this.

1. I agree with you, introducing this feature can be the occasion to also close #3722 by considering that masked areas are `0` labelled.
2. segments dtype did not change, it is still `double`. `bint` for the mask seems a better choice. The `mask.size` test is indeed cleaner.
3. The `only_dist` parameter is for updating centroids positions without considering pixels color (it is the step 2 in the maskSLIC algorithm). It makes me think that renaming the parameter `ignore_color` is may be more clear.
4. `regular_grid` reimplementation is just a matter of code style and code lines count reduction :-).

Concerning the last dtype modification that you point, labeling masked area with `-1` in the current implementation prevents the use of the unsigned `Py_size_t`.

I will tak eyour remarks into consideration and come with an other proposition soon.
Thanks again.

Reply:
FYI: Py_ssize_t = signed size type

https://www.python.org/dev/peps/pep-0353/#specification

Reply:
THX, the double `s` makes sens then :-)

Reply:
If that makes sense to you, maybe you can help me make sense of these 
https://docs.microsoft.com/en-us/cpp/c-runtime-library/reference/snprintf-s-snprintf-s-l-snwprintf-s-snwprintf-s-l?view=vs-2019

Reply:
:-)